packages/vite/src/node/plugins/define.ts

```diff
@@ -49,16 +57,20 @@ export function definePlugin(config: ResolvedConfig): Plugin {
   const importMetaKeys: Record<string, string> = {}
   const importMetaFallbackKeys: Record<string, string> = {}
   if (isBuild) {
-    env.SSR = !!config.build.ssr
-
+    const env: Record<string, any> = {
+      ...config.env,
+      SSR: !!config.build.ssr,
+    }
     // set here to allow override with config.define
     importMetaKeys['import.meta.hot'] = `undefined`
     for (const key in env) {
       importMetaKeys[`import.meta.env.${key}`] = JSON.stringify(env[key])
     }
     Object.assign(importMetaFallbackKeys, {
       'import.meta.env.': `({}).`,
-      'import.meta.env': JSON.stringify(env),
+      'import.meta.env': JSON.stringify({ ...env, ...userDefineEnv })
+        .replace(/"__vite__/g, '')
```

Comment:
Maybe we should use `/"__vite__(.+)?__vite__"`, in case there some extra values only starting `"__vite__`  or trailing `__vite__"`

Reply:
Good idea. I've replaced it with `/"__vite__(.+?)__vite__"/` as I think that's more accurate 🤔